### PR TITLE
Integration tests for Kubeflow Bundling with ACK and KFP Components

### PR DESCRIPTION
**Pull request by mbaijal**
**Description of your changes:**
Added two integration tests for Kubeflow Bundling -
1. A Simple ACK test which creates a profile, spins up a notebook and checks that the trainingjob CRD is installed
2. A KFP test which runs the sample SageMaker training pipeline and makes sure it is successful - also uses profile.

**TODO**:
Permissions are broader than they need to be at the moment. The SageMaker and S3 accesses are added to both Roles, can be streamlined. 

**Testing:**
- [ ] Unit tests pass
- [ ] e2e tests pass
- Details about new tests (If this PR adds a new feature)
- Details about any manual tests performed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/e2e/fixtures/profile_dependencies.py [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        ack_yaml_original = unmarshal_yaml(yaml_file=ack_filename)
+        ack_yaml = unmarshal_yaml(
+            yaml_file=ack_filename,
+            replacements={"ACK_SAGEMAKER_OIDC_ROLE": oidc_role_arn},
+        )
+
+        with open(ack_filename, "w") as file:
+            file.write(str(yaml.dump(ack_yaml)))
+
+        # # 3. Edit Params.env
```
Comment (mbaijal) [inline tests/e2e/fixtures/profile_dependencies.py:226]: remove after confirming @mbaijal

File: tests/e2e/tests/test_ack.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+NOTEBOOK_IMAGES = [
+    "public.ecr.aws/c9e4w0g3/notebook-servers/jupyter-tensorflow:2.6.3-cpu-py38-ubuntu20.04-v1.8",
+]
+
+testdata = [
+    ("ack", NOTEBOOK_IMAGES[0], "verify_ack_integration.ipynb", "No resources found in kubeflow-user-example-com namespace"),
+]
+
+# GENERIC_KUSTOMIZE_MANIFEST_PATH = "../../deployments/vanilla"
```
Comment (mbaijal) [inline tests/e2e/tests/test_ack.py:42]: cleanup @mbaijal

File: tests/e2e/tests/test_ack.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+# @pytest.fixture(scope="class")
+# def kustomize_path():
+#     return GENERIC_KUSTOMIZE_MANIFEST_PATH
+
+class TestACK:
```
Comment (surajkota) [inline tests/e2e/tests/test_ack.py:48]: need to add the the test as part of to test_sanity_portforward, it needs to run in canaries

Comment (mbaijal) [reply tests/e2e/tests/test_ack.py:48]: oh is that the only way to run it in canaries ? And do we need to add both tests - ack and kfp or just one?

File: tests/e2e/utils/s3_for_training/data_bucket.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        except ClientError:
+            logger.exception(f"failed to create S3 bucket {self.name}")
+            raise
+        else:
+            logger.info(f"created s3 bucket {self.name}.")
+            return self.name
+
+    def delete(self):
+        try:
+            self.s3_client.delete_bucket(Bucket=self.name)
```
Comment (rrrkharse) [inline tests/e2e/utils/s3_for_training/data_bucket.py:58]: does this work if the bucket has objects in it?

File: tests/e2e/fixtures/profile_dependencies.py [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+        # ack_filename = TO_ROOT_PATH + "awsconfigs/common/ack-sagemaker-controller/params.env"
+        # ack_yaml_original = unmarshal_yaml(yaml_file=ack_filename)
+        # ack_yaml = unmarshal_yaml(
+        #     yaml_file=ack_filename,
+        #     replacements={"ACK_SAGEMAKER_OIDC_ROLE": oidc_role_arn},
+        # )
+
+        # with open(ack_filename, "w") as file:
+        #     file.write(str(yaml.dump(ack_yaml)))
```
Comment (rrrkharse) [inline tests/e2e/fixtures/profile_dependencies.py:236]: nit: remove

File: tests/e2e/fixtures/profile_dependencies.py [hunk 6]
```diff
@@ -0,0 +1,250 @@
...
+
+    def on_delete():
+        pass
+        # with open(filename, "w") as file:
+        #     file.write(str(yaml.dump(profile_yaml_original)))
```
Comment (rrrkharse) [inline tests/e2e/fixtures/profile_dependencies.py:241]: nit: remove

File: tests/e2e/tests/test_ack.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    host,
+    password,
+)
+from e2e.fixtures.notebook_dependencies import notebook_server
+
+# GENERIC_KUSTOMIZE_MANIFEST_PATH = "../../deployments/vanilla"
+
+# @pytest.fixture(scope="class")
+# def kustomize_path():
+#     return GENERIC_KUSTOMIZE_MANIFEST_PATH
```
Comment (rrrkharse) [inline tests/e2e/tests/test_ack.py:35]: nit: remove

**Review by rrrkharse — APPROVED** [review]

File: tests/e2e/fixtures/profile_dependencies.py [hunk 8]
```diff
@@ -0,0 +1,250 @@
...
+CUSTOM_RESOURCE_TEMPLATES_FOLDER = "./resources/custom-resource-templates"
+
+
+@pytest.fixture(scope="class")
+def kustomize_path():
+    return TO_ROOT_PATH + "tests/e2e/resources/custom-manifests/profile-irsa"
+
+
+@pytest.fixture(scope="class")
+def profile_controller_policy(region, metadata, request):
```
Comment (surajkota) [inline tests/e2e/fixtures/profile_dependencies.py:38]: can you summarize what is happening with profile controller in this test?

File: tests/e2e/fixtures/profile_dependencies.py [hunk 9]
```diff
@@ -0,0 +1,250 @@
...
+        request=request,
+        resource_details="created",
+        metadata_key=metadata_key,
+        on_create=on_create,
+        on_delete=on_delete,
+    )
+
+
+@pytest.fixture(scope="class")
+def profile_trust_policy(
```
Comment (surajkota) [inline tests/e2e/fixtures/profile_dependencies.py:113]: I think we should use the setup_ack script directly, lets get that pr merged

**Comment by mbaijal** [comment]
Closing this PR, creating a new one since there have been significant changes and needed a new branch.
